### PR TITLE
Adapt perl script to handle multiple .desc files (test-gen-support edition)

### PR DESCRIPTION
**Pull request by LAJW**
Changes test.pl to name output *.out files after *.desc instead
of *.class files. Also updates failed-test-printer.pl to work with
those new *.out files and multiple *.desc files.

### CODE REVIEW

**Comment by LAJW** [comment]
This change is already merged with master. To make this functionality available to test-gen teams as soon as possible I was asked to rebase and create PR again.

Issue: https://github.com/diffblue/test-gen/issues/680#issuecomment-317952761

File: regression/test.pl [hunk 1]
```diff
@@ -71,7 +71,9 @@ ($$$$$)
 
   $options =~ s/$ign//g if(defined($ign));
 
-  my $output = $input;
+  my $descriptor = $test;
+  $descriptor =~ s/^.*\///;
```
Comment (smowton) [inline regression/test.pl]: Add end-of-line comment indicating what this is doing ("remove '/' characters"?)

Comment (LAJW) [inline regression/test.pl]: I'm going to draw a line here and say: no comments needed: This is a pretty straightforward Regexp string replace. Replaces "^.*\/" with an empty string. Which mans converts path to basename.

This is _the_ standard pattern for doing any kind of string replacement in perl. I know it looks arcane at first glance, but so does putting argument list into a sub. File IO on the other hand...

Comment (allredj) [inline regression/test.pl]: I agree with @smowton here that a comment `convert to path to basename` wouldn't hurt.

Comment (tautschnig) [inline regression/test.pl]: ... or `use File::Basename;` and do `my $descriptor = basename($test);` and have self-explanatory code.

Comment (smowton) [inline regression/test.pl]: @LAJW not everyone reads Perl, and I can't really Google that string of punctuation, can I? I've an inkling that it's probably related to `sed`'s `s///` operator, but why make me slog through that process for a quarter hour when you could tell me what you're doing in 10 seconds?

Comment (LAJW) [inline regression/test.pl]: @smowton I'm in minority here, and so I've already force-pushed the File::Basename version, which is of course clearer. What I was basing of comments on is that it's not the first instance of that of that kind of expression in the perl files we have so I assumed it a norm.

**Comment by smowton** [comment]
@LAJW next make the corresponding test-gen PR that updates the cbmc pointer to point to the tip of this branch, to exhibit that the changes don't break any of the test-gen tests (which directly use this script)

**Comment by LAJW** [comment]
PR on test-gen: https://github.com/diffblue/test-gen/pull/841

**Pull request merged**